### DISCUSSION TITLE
Signature box coordinate calculation on rotated page

### DISCUSSION THREAD

**Discussion by ihegde** [MatthiasValvekens/pyHanko#156] [Q&A] [answered]
Hi, 
Thanks for including the fields.VisibleSigSettings in the new release. I'm using it as below: 

    w = IncrementalPdfFileWriter(inf)
    fields.append_signature_field(
        w, sig_field_spec=fields.SigFieldSpec(
            'Signature', 
            visible_sig_settings=fields.VisibleSigSettings(rotate_with_page=False),
            box=(0, 0, 25, 25) 
        )
    )

The signature works, I tried a few coordinates but unable to get a hang of the position and size calculations. 
I understand the coordinates are LLx, LLy, URx, URy ( LL: Lower Left, UR, Upper right) 

Now with rotated axis, X is the vertical axis and Y horizontal, and the origin is on top left corner. 
So I'm expecting the Lower left to become upper left, and upper right to become lower right. 
With the above coordinates (0, 0, 25, 25) I was expecting it to draw on top left corner, with 25 width and height. But it comes up as (image attached). I've tried many other coordinates, just giving this as sample since it is easy to understand when we have one of them as 0, 0. How to interpret this?  My goal is to find a text and add signature below that, But I'm unable to translate the text coordinates to signature coordinates. 
![PDF sig](https://user-images.githubusercontent.com/25707247/191018203-4647d56b-6647-49e5-914e-41a5e0af4ab0.png)

**Comment by MatthiasValvekens** [comment, marked as answer]
Hi @ihegde,


The answer to your question is in the specification (clause 12.5.3 in either ISO 32000-2:2020 or ISO 32000-1:2008), which has the following to say about _NoZoom_ and _NoRotate_:

 
> [...] the annotation’s position is defined by the coordinates of the upper-left corner of its annotation rectangle, as defined by the Rect entry in the annotation dictionary and interpreted in the default user space of the page. When the default user space is scaled or rotated, the positions of the other three corners of the annotation rectangle are different in the altered user space than they were in the original user space. The PDF processor shall perform this alteration automatically. However, it shall not actually change the annotation’s Rect entry, which continues to describe the annotation’s relationship with the unscaled, unrotated user space.


In other words, the annotation "pivots" around its top left corner, as it were, and you specify the annotation's dimensions in the original, unrotated coordinate system. There's an illustrative figure in the standard as well.

Here's a link to the freely available version of ISO 32000-1:2008, if you want to look up the text I cited: https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf. Both the explanation and the figure are in clause 12.5.3.

**Reply by ihegde** [reply]
@MatthiasValvekens 
Thank you for the pointers. Understood. Will figure out the translation from here.
